lib/daru/dataframe.rb

```diff
@@ -2177,5 +2276,64 @@ def one_to_many_row row, number, vars, pattern
           [v, row[name]]
         }.to_h
     end
+
+    # Raises IndexError when one of the positions is a valid position
+    def validate_vector_positions *positions
+      positions = [positions] if positions.is_a? Integer
+      positions.each do |pos|
+        raise IndexError, "#{pos} is not a valid position." if pos >= shape[1]
+      end
+    end
+
+    # Raises IndexError when one of the positions is a valid position
+    def validate_row_positions *positions
+      positions = [positions] if positions.is_a? Integer
+      positions.each do |pos|
+        raise IndexError, "#{pos} is not a valid position." if pos >= shape[0]
+      end
+    end
+
+    # Accepts hash, enumerable and vector and align it properly so it can be added
+    def preprocess_vector vector
+      case vector
+      when Daru::Vector
+        vector.reindex @vectors
+      when Hash
+        Daru::Vector.new(vector).reindex @vectors
+      else
+        Daru::Vector.new vector
+      end
+    end
+
+    # Preprocess ranges, integers and array in appropriate ways
+    def preprocess_row_positions *positions
+      if positions.size == 1
+        case positions.first
+        when Integer
+          positions.first
+        when Range
+          shape[0].times.to_a[positions.first]
+        else
+          raise ArgumentError, 'Unkown position type.'
+        end
+      else
+        positions
+      end
+    end
+
+    def preprocess_vector_positions *positions
+      if positions.size == 1
+        case positions.first
+        when Integer
+          positions.first
+        when Range
+          shape[1].times.to_a[positions.first]
+        else
+          raise ArgumentError, 'Unkown position type.'
+        end
+      else
+        positions
+      end
+    end
```

Comment:
I am thinking of creating one method `preprocess_positions` and then passing symbol `:row` or `:vector`. Or should I do something else?